plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -80,192 +197,724 @@ function executeCLI($scriptName, &$args) {
+		}
+
+		if ($failedRows === 0) {
+			echo __('plugin.importexport.csv.allDataSuccessfullyImported', ['processedRows' => $processedRows]) . "\n\n";
+			unlink($csvForInvalidRowsName);
+		} else {
+			echo __('plugin.importexport.csv.seeInvalidRowsFile', ['processedRows' => $processedRows - $failedRows, 'failedRows' => $failedRows]) . "\n\n";
+		}
+	}
+
+	/**
+	 * Display the command-line usage information
+	 */
+	function usage($scriptName) {
+		echo __('plugins.importexport.csv.cliUsage', [
+			'scriptName' => $scriptName,
+			'pluginName' => $this->getName()
+		]) . "\n";
+	}
+
+	/**
+	 * Parse and validate initial command args
+	 *
+	 * @param string $scriptName
+	 * @param array $args
+	 * @return string[]
+	 */
+	private function parseCommandLineArguments($scriptName, $args) {
 		$filename = array_shift($args);
 		$username = array_shift($args);
+		$basePath = dirname($filename);
 
 		if (!$filename || !$username) {
 			$this->usage($scriptName);
-			exit();
+			exit(1);
+		}
+
+		return [$filename, $username, $basePath];
+	}
+
+	/**
+	 * Retrieve and validate the User by username
+	 *
+	 * @param string $username
+	 */
+	private function validateUser($username) {
+		$user = $this->getUser($username);
+		if (!$user) {
+			echo __('plugins.importexport.csv.unknownUser', ['username' => $username]) . "\n";
+			exit(1);
+		}
+
+		$this->_userId = $user->getId();
+	}
+
+	/**
+	 * @param string $filename
+	 * @param 'r'|'w' $mode
+	 */
+	private function createNewFile($filename, $mode) {
+		try {
+			return new SplFileObject($filename, $mode);
+		} catch (Exception $e) {
+			echo $e->getMessage();
+			exit(1);
+		}
+	}
+
+	/**
+	 * @param string $filename
+	 */
+	private function createAndValidateCSVFile($filename) {
+		$file = $this->createNewFile($filename, 'r');
+		$file->setFlags(SplFileObject::READ_CSV);
+
+		$headers = $file->fgetcsv();
+
+		$missingHeaders = array_diff($this->_expectedHeaders, $headers);
+
+		if (count($missingHeaders)) {
+			echo __('plugin.importexport.csv.missingHeadersOnCsv', ['missingHeaders' => $missingHeaders]);
+			exit(1);
+		}
+
+		return $file;
+	}
+
+	/**
+	 * @param string $basePath
+	 */
+	private function createAndValidateCSVFileInvalidRows($csvForInvalidRowsName) {
+		$invalidRowsFile = $this->createNewFile($csvForInvalidRowsName, 'w');
+		$invalidRowsFile->fputcsv(array_merge(array_pad($this->_expectedHeaders, $this->_expectedRowSize, null), ['error']));
+
+		return $invalidRowsFile;
+	}
+
+	/**
+	 * Insert static data that will be used for the submission processing
+	 */
+	private function initializeStaticVariables() {
+		$this->_dirNames = Application::getFileDirectories();
+		$this->_format = trim($this->_dirNames['context'], '/') . '/%d/' . trim($this->_dirNames['submission'], '/') . '/%d';
+
+		$this->_fileManager = new FileManager();
+		$this->_publicFileManager = new PublicFileManager();
+
+		$this->_fileService = Services::get('file');
+		$this->_publicationService = Services::get('publication');
```

Comment:
These things are cheap to initialize, but I don't see any problem to cache them (especially if you're using in a lot of places), so I'm just leaving a note.